src/internal/connector/onedrive/item.go

```diff
@@ -410,33 +411,81 @@ func constructWebURL(adtl map[string]any) string {
 	return url
 }
 
-// func fetchParentReference(
-// 	ctx context.Context,
-// 	service graph.Servicer,
-// 	orig models.ItemReferenceable,
-// ) (models.ItemReferenceable, error) {
-// 	if orig == nil || service == nil || ptr.Val(orig.GetName()) != "" {
-// 		return orig, nil
-// 	}
+func fetchDriveName(
+	ctx context.Context,
+	service graph.Servicer,
+	driveID string,
+) (string, error) {
+	if service == nil || driveID == "" {
+		return "", nil
+	}
 
-// 	options := &msdrives.DriveItemRequestBuilderGetRequestConfiguration{
-// 		QueryParameters: &msdrives.DriveItemRequestBuilderGetQueryParameters{
-// 			Select: []string{"name"},
-// 		},
-// 	}
+	options := &msdrives.DriveItemRequestBuilderGetRequestConfiguration{
+		QueryParameters: &msdrives.DriveItemRequestBuilderGetQueryParameters{
+			Select: []string{"name"},
+		},
+	}
 
-// 	driveID := ptr.Val(orig.GetDriveId())
+	drive, err := service.Client().DrivesById(driveID).Get(ctx, options)
+	if err != nil {
+		return "", graph.Stack(ctx, err)
+	}
 
-// 	if driveID == "" {
-// 		return orig, nil
-// 	}
+	return *drive.GetName(), nil
+}
 
-// 	drive, err := service.Client().DrivesById(driveID).Get(ctx, options)
-// 	if err != nil {
-// 		return nil, graph.Stack(ctx, err)
-// 	}
+func updateParentReference(
+	ctx context.Context,
+	service graph.Servicer,
+	orig models.ItemReferenceable,
+	driveMap *map[string]string,
+	mu *sync.Mutex,
+) (models.ItemReferenceable, error) {
+	if orig == nil {
+		mu.Unlock()
+		return nil, nil
+	}
 
-// 	orig.SetName(drive.GetName())
+	driveID, ok := ptr.ValOK(orig.GetDriveId())
+	if !ok {
+		mu.Unlock()
+		return orig, nil
+	}
 
-// 	return orig, nil
-// }
+	var (
+		drives      = *driveMap
+		name, inMap = drives[driveID]
+		err         error
+	)
+
+	if inMap {
+		mu.Unlock()
+	} else {
+
+		name, err = fetchDriveName(ctx, service, driveID)
+		if err != nil {
+			return nil, err
+		}
+
+		drives[driveID] = name
+		mu.Unlock()
+		logger.Ctx(ctx).Infof("Drive Name not in given map: Retrieved: %s", name)
+	}
+
+	orig.SetName(&name)
+
+	return orig, nil
+}
```

Comment:
It's not clear to me why we're calling `fetchDriveName` here? Don't we already have the drive name in the driveMap?

Reply:
@dadams39 and I synced 1:1 on this. We should be able to just stash the driveName in the `onedrive.Collection` and use it issuing any other queries.

Reply:
A bug within `UpdateCollections` caused incorrect population of collections when multiple drives were involved.  Fixes within [f87066e](https://github.com/alcionai/corso/pull/2733/commits/f87066ec3f9062f3ee02f159c15c820c83b1b99d)